internal/stage_alternation.go

```diff
@@ -19,16 +19,6 @@ func testAlternation(stageHarness *test_case_harness.TestCaseHarness) error {
 			Input:            "a cow",
 			ExpectedExitCode: 1,
 		},
-		{
-			Pattern:          "^I see (\\d (cat|dog|cow)s?(, | and )?)+$",
```

Comment:
I don't think removing all combination cases is the answer here @andy1li - we regularly get people complaning about how our tests aren't exhaustive enough and don't cover enough cases. Can we find a set of tests that support alternation and are slightly complex / combined with others, but don't use nested grouping in general?